winpr/libwinpr/path/shell.c

```diff
@@ -58,6 +60,24 @@ static char* GetPath_XDG_RUNTIME_DIR(void);
  * http://msdn.microsoft.com/en-us/library/windows/desktop/bb762188/
  */
 
+#if defined(WIN32) && !defined(_UWP)
+
+static char* win_get_known_folder(REFKNOWNFOLDERID id, BOOL system)
```

Comment:
The `system` parameter is confusing as -1 indicates the "Default User": https://learn.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shgetknownfolderpath

Since this is only used once for `FOLDERID_ProgramData` (and in that case for no obvious reason) we should remove it and just use `NULL` for the handle parameter.

Reply:
well, the `obvious` reason is that we do want the system install program data folder, not the user specific one ;)